codes/c/chapter_divide_and_conquer/hanota.c

```diff
@@ -0,0 +1,76 @@
+/**
+ * File: hanota.c
```

Comment:
Hi @Zuoxun , could you create an another PR for this file? It will help make each PR have clear boundaries.

Reply:
I have already created 3 PRs for 3 files titled "vector.h", "hanota.c", "binary_search_recur.c" now. Please review it.